Sources/VaporJWT/AuthError.swift

```diff
@@ -0,0 +1,89 @@
+/// Errors thrown during the parsing, creation
+/// and verification of JWT tokens by this package.
+public struct AuthError: Swift.Error {
+    public let code: Int
+    public let reasonPhrase: String
+    public let origin: Swift.Error?
+
+    init(code: Int, reasonPhrase: String, origin: Swift.Error?) {
+        self.code = code
+        self.reasonPhrase = reasonPhrase
+        self.origin = origin
+    }
+
+    public static func noAuthorizationHeader(_ origin: Swift.Error? = nil) -> Error {
```

Comment:
I'm open to either, this seems like as reasonable solution as any for the time being at least! Enums would be ideal, but they cost too much for now and this recreates as best as we can. I wish we could enforce exhaustive _and_ default 🤔